Sync/Synchronizers/Bookmarks/BookmarksSynchronizer.swift

```diff
@@ -126,6 +126,12 @@ open class BufferingBookmarksSynchronizer: TimestampedSingleCollectionSynchroniz
                     return buffer.validate().bind { validationResult in
                         if let invalidError = validationResult.failureValue as? BufferInvalidError {
                             self.statsSession.validationStats = self.validationStatsFrom(error: invalidError)
+
+                            log.warning("Buffer inconsistent, starting repair procedure")
```

Comment:
Let's put all this behind a new flag in `AppConstants`, and add Sentry logging to record each time when we run a repair. The ideal is that we only validate when we've downloaded new data, and we only repair after we validate, and so we rarely repair.